engine/context/src/main/java/io/deephaven/engine/context/ExecutionContext.java

```diff
@@ -112,7 +104,21 @@ public ExecutionContext withSystemic(boolean isSystemic) {
         if (isSystemic == this.isSystemic) {
             return this;
         }
-        return new ExecutionContext(isSystemic, queryLibrary, queryScope, queryCompiler);
+        return new ExecutionContext(isSystemic, authContext, queryLibrary, queryScope, queryCompiler);
+    }
+
+    /**
+     * Returns, or creates, an execution context with the given value for {@code authContext} and existing values for
+     * the other members. This is not intended to be used by user code.
+     *
+     * @param authContext the auth context to use instead; null values are ignored
+     * @return the execution context
+     */
+    public ExecutionContext withAuthContext(final AuthContext authContext) {
+        if (authContext == null || authContext == this.authContext) {
+            return this;
+        }
```

Comment:
The check intends to prevent the user from setting it null. It's kind of arbitrary. As long as DHE doesn't assume a null auth context is the same as the instance auth context. The user would only be able to change it while a script is being executed, or from a spawned thread.